src/games/strategy/triplea/ui/menubar/ExportMenu.java

```diff
@@ -168,27 +173,48 @@ public String getDescription() {
       // A small warning so users will not over-write a file,
       if (f.exists()) {
         final int choice =
-            JOptionPane.showConfirmDialog(null, "A file by that name already exists. Do you wish to over write it?",
+            JOptionPane.showConfirmDialog(frame, "A file by that name already exists. Do you wish to over write it?",
                 "Over-write?", JOptionPane.YES_NO_OPTION, JOptionPane.WARNING_MESSAGE);
         if (choice != JOptionPane.OK_OPTION) {
           return;
         }
       }
+
       final File file = f;
-      final Runnable t = () -> {
-        if (saveScreenshot(node, file, frame, gameData)) {
-          JOptionPane.showMessageDialog(null, "Map Snapshot Saved", "Map Snapshot Saved",
-              JOptionPane.INFORMATION_MESSAGE);
+      final SwingWorker<?, ?> task = new SwingWorker<Void, Void>() {
```

Comment:
Sounds good.  I took a brief look at `SwingComponents` while working this PR for an unrelated reason and didn't make the connection for it being a good place to extract a generic progress runner.

@DanVanAtta Did you envision `SaveScreenshot` (and all the code related to it) still being internal to `ExportMenu`?  Or would it be preferable to extract it to its own file in a different package?  If external, can you suggest an appropriate package, please?